File: frontend/src/types/quotation.ts
export type QuotationStatus = 'draft' | 'sent' | 'approved' | 'rejected'

export interface CustomerBrief {
  id: number
  first_name: string
  last_name: string
  company: string | null
}

export interface ForkliftBrief {
  id: number
  serial_number: string
  name_en: string
}

export interface UserBrief {
  id: number
  username: string
  full_name: string | null
}

export interface Quotation {
  id: number
  quotation_no: string
  customer: CustomerBrief
  forklift: ForkliftBrief | null
  expected_start_date: string
  expected_end_date: string
  rental_price: number
  daily_hours_quota: number
  status: QuotationStatus
  notes: string | null
  creator: UserBrief | null
  created_at: string
  updated_at: string | null
}

export interface QuotationListResponse {
  items: Quotation[]
  total: number
  page: number
  page_size: number
  pages: number
}

export interface QuotationCreate {
  customer_id: number
  forklift_id?: number | null
  expected_start_date: string
  expected_end_date: string
  rental_price: number
  daily_hours_quota?: number
  status?: QuotationStatus
  notes?: string
}

export interface QuotationUpdate {
  customer_id?: number
  forklift_id?: number | null
  expected_start_date?: string
  expected_end_date?: string
  rental_price?: number
  daily_hours_quota?: number
  status?: QuotationStatus
  notes?: string
}

export interface QuotationListParams {
  q?: string
  status?: QuotationStatus
  customer_id?: number
  page?: number
  page_size?: number
}

// Payload handed from QuotationDetailPage to the rental contract editor via
// router state when converting an approved quotation into a contract.
export interface QuotationConversionPrefill {
  quotation_id: number
  quotation_no: string
  customer_id: number
  forklift_id: number | null
  expected_start_date: string
  expected_end_date: string
  rental_price: number
  daily_hours_quota: number
}

File: backend/app/schemas/quotation.py
from datetime import date, datetime

from pydantic import BaseModel, Field

from app.models.quotation import QuotationStatus


# ── Brief models ─────────────────────────────────────────────────────────────

class CustomerBrief(BaseModel):
    model_config = {"from_attributes": True}
    id: int
    first_name: str
    last_name: str
    company: str | None = None


class ForkliftBrief(BaseModel):
    model_config = {"from_attributes": True}
    id: int
    serial_number: str
    name_en: str


class UserBrief(BaseModel):
    model_config = {"from_attributes": True}
    id: int
    username: str
    full_name: str | None = None


# ── Quotation schemas ────────────────────────────────────────────────────────

class QuotationCreate(BaseModel):
    customer_id: int
    forklift_id: int | None = None
    expected_start_date: date
    expected_end_date: date
    rental_price: float = Field(..., ge=0)
    daily_hours_quota: int = Field(default=8, gt=0, le=24)
    status: QuotationStatus = QuotationStatus.DRAFT
    notes: str | None = None


class QuotationUpdate(BaseModel):
    customer_id: int | None = None
    forklift_id: int | None = None
    expected_start_date: date | None = None
    expected_end_date: date | None = None
    rental_price: float | None = Field(default=None, ge=0)
    daily_hours_quota: int | None = Field(default=None, gt=0, le=24)
    status: QuotationStatus | None = None
    notes: str | None = None


class QuotationResponse(BaseModel):
    model_config = {"from_attributes": True}
    id: int
    quotation_no: str
    customer: CustomerBrief
    forklift: ForkliftBrief | None = None
    expected_start_date: date
    expected_end_date: date
    rental_price: float
    daily_hours_quota: int
    status: str
    notes: str | None = None
    creator: UserBrief | None = None
    created_at: datetime
    updated_at: datetime | None = None


class QuotationListResponse(BaseModel):
    items: list[QuotationResponse]
    total: int
    page: int
    page_size: int
    pages: int
